cmd/utils.go

```diff
@@ -180,14 +213,62 @@ func parseBuildInfoJSON(jsonFile, spdxID string) (sigURL, gcpKmsKey string, buil
     fields := strings.Split(gcpKmsKey, "/")
     for index, str := range fields {
         if str == "cryptoKeys" {
-            gcpKmsKey = fields[index + 1]
+            gcpKmsKey = fields[index+1]
             break
         }
     }
 
-    return  sigURL, gcpKmsKey, buildProvSig, nil
+    return sigURL, gcpKmsKey, buildProvSig, nil
 }
 
+func parsePremiumBuildInfoJSON(jsonFile, suffix string) (sigDetails SigDetails, provenancePublicKey string, buildProvSig []byte, err error) {
+    // Read the JSON file.
+    data, err := os.ReadFile(jsonFile)
+    if err != nil {
+        return SigDetails{}, "", nil, fmt.Errorf("failed to read JSON file: %v", err)
+    }
+
+    var amalgamView *amalgamView
+    if err = json.Unmarshal(data, &amalgamView); err != nil {
+        return SigDetails{}, "", nil, fmt.Errorf("failed to unmarshal JSON data: %v", err)
+    }
+
+    var jsonData *premiumBuildInfo
+    if err = json.Unmarshal([]byte(amalgamView.BuildInfo), &jsonData); err != nil {
+        return SigDetails{}, "", nil, fmt.Errorf("failed to unmarshal buildInfo JSON data: %v", err)
+    }
+    var sbomData *sbom
+    if err = json.Unmarshal([]byte(jsonData.Sbom), &sbomData); err != nil {
+        return SigDetails{}, "", nil, fmt.Errorf("failed to unmarshal 'sbom' data %v: %v", jsonData.Sbom, err)
+    }
+
+    // Get signature Details of the package.
+    for _, element := range sbomData.Packages {
+        if strings.HasSuffix(element.Spdxid, suffix) {
+            comment := element.Annotations[0].Comment
+            if err = json.Unmarshal([]byte(comment), &sigDetails); err != nil {
+                return SigDetails{}, "", nil, fmt.Errorf("failed to unmarshal JSON data: %v", err)
+            }
+        }
+    }
+
+    // Get build provenance signatures and public key.
+    if len(jsonData.BuildDetails) < 1 {
+        return SigDetails{}, "", nil, fmt.Errorf("couldn't get build details")
+    }
+    if len(jsonData.BuildDetails[0].BuildProvenances) < 1 {
+        return SigDetails{}, "", nil, fmt.Errorf("couldn't get build provenance details")
```

Comment:
we can make sigdetails as pointer, and send it as nil

Reply:
done